e2e/playwright/search/src/tests/search-filters-logic.spec.ts

```diff
@@ -0,0 +1,125 @@
+    try {
+      const apiClientFactory = new ApiClientFactory();
+      await apiClientFactory.setUpAcaBackend('admin');
+      await apiClientFactory.createUser({ username });
+      nodesApi = await NodesApi.initialize(username, username);
+      trashcanApi = await TrashcanApi.initialize(username, username);
+      fileActionsApi = await FileActionsApi.initialize(username, username);
+      await nodesApi.createFile(logicFile1);
+      await nodesApi.createFile(logicFile2);
+      await fileActionsApi.waitForNodes(logicFile1, { expect: 1 });
+      await fileActionsApi.waitForNodes(logicFile2, { expect: 1 });
+    } catch (error) {
+      console.error(`beforeAll failed: ${error}`);
+    }
+  });
+
+  test.afterAll(async () => {
+    await Utils.deleteNodesSitesEmptyTrashcan(nodesApi, trashcanApi, 'afterAll failed');
+  });
+
+  test('[C699500] Filter with Match All', async ({ searchPage }) => {
+    await searchPage.searchFilters.logicFilter.click();
+    await searchPage.searchFiltersLogic.matchAllInput.fill(
+      `${logicFile1Split[0]} ${logicFile1Split[1]} ${logicFile1Split[2]} ${logicFile1Split[3]} ${logicFile1Split[4]}`
+    );
+    await searchPage.searchFiltersLogic.applyButton.click();
+    await searchPage.dataTable.progressBarWaitForReload();
+
+    await expect(searchPage.dataTable.getRowByName(logicFile1)).toBeVisible();
+    await expect(searchPage.dataTable.getRowByName(logicFile2)).not.toBeVisible();
+  });
+
+  test('[C699501] Filter with Match Any', async ({ searchPage }) => {
+    await searchPage.searchFilters.logicFilter.click();
+    await searchPage.searchFiltersLogic.matchAnyInput.fill(`${logicFile1Split[1]}-${logicFile1Split[2]}-${logicFile1Split[3]} ${logicFile1Split[4]}`);
+    await searchPage.searchFiltersLogic.applyButton.click();
+    await searchPage.dataTable.progressBarWaitForReload();
+
+    expect(await searchPage.dataTable.getRowsCount()).toBe(3);
+    await expect(searchPage.dataTable.getRowByName(logicFile1)).toBeVisible();
+    await expect(searchPage.dataTable.getRowByName(logicFile2)).toBeVisible();
+  });
+
+  test('[C699502] Filter with Exclude', async ({ searchPage }) => {
+    await searchPage.searchFilters.logicFilter.click();
+    await searchPage.searchFiltersLogic.matchAnyInput.fill(`${logicFile1Split[1]}-${logicFile1Split[2]} ${logicFile1Split[4]}`);
+    await searchPage.searchFiltersLogic.excludeInput.fill(`${logicFile1Split[3]}`);
+    await searchPage.searchFiltersLogic.applyButton.click();
+    await searchPage.dataTable.progressBarWaitForReload();
+
+    expect(await searchPage.dataTable.getRowsCount()).toBe(2);
+    await expect(searchPage.dataTable.getRowByName(logicFile1)).not.toBeVisible();
+    await expect(searchPage.dataTable.getRowByName(logicFile2)).toBeVisible();
+  });
+
+  test('[C699503] Filter with Exact phrase', async ({ searchPage }) => {
+    await searchPage.searchFilters.logicFilter.click();
+    await searchPage.searchFiltersLogic.matchExactInput.fill(logicFile1);
```

Comment:
The exact should test the phrase that name, title or description contains rather than whole name of the file imo